### PR TITLE
Changes preprocessing function in emotion-fer-plus

### PR DESCRIPTION
**Pull request by ahmedhosny**
I have reason to believe the preprocessing function is incorrect. It seems to be ignoring the channels in the image and the results are off. With reference to the onnx.js implementation of the same model [here](https://github.com/Microsoft/onnxjs-demo/blob/master/src/components/models/Emotion.vue#L48), they seem to be first grayscaling the image if indeed it has 3/4 channels, then they are centering the values around 0 by subtracting 127.5 and dividing by the same. I am proposing to change the preprocessing function. Currently written for clarity, feel free to shorten. I am working on an [implementation](http://app.modelhub.ai/) and found that the onnx.js preprocessing gives more realistic results (they used the HTML canvas to do the resizing to 64x64, so there is really no concrete way to fully reproduce their implementation outside www and benchmark against it).

# Model Name

## Use-cases
Model use cases

## Description
Description of model

## Model

 |Model        |Download  |Checksum| Download (with sample test data)|ONNX version|Opset version|Accuracy |
|-------------|:--------------|:--------------|:--------------|:--------------|:--------------|:--------------|
|Model Name       |ONNX Model download link with size  | MD5 checksum for the ONNX model| tar file containing ONNX model and synthetic test data (in .pb format)|ONNX version used for conversion| Opset version used for conversion|Accuracy values |

All ONNX models must pass the ONNX model checker before contribution. The snippet of code below can be used to perform the check. If any errors are encountered, it implies the check has failed.

```
import onnx
from onnx import checker
model_proto = onnx.load("path to .onnx file")
checker.check_model(model_proto)
```

<hr>

## Inference
Step by step instructions on how to use the pretrained model and link to an example notebook/code. This section should ideally contain:

### Input to model
Input to network (Example: 224x224 pixels in RGB)

### Preprocessing steps
Preprocessing required

### Output of model
Output of network

### Postprocessing steps
Post processing and meaning of output
<hr>

## Dataset (Train and validation)
This section should discuss datasets and any preparation steps if required.
<hr>

## Validation accuracy
Details of experiments leading to accuracy and comparison with the reference paper.
<hr>

## Training
Training details (preprocessing, hyperparameters, resources and environment) along with link to a training notebook (optional). 

Also clarify in case the model is not trained from scratch and include the source/process used to obtain the ONNX model.
<hr>

## Validation
Validation script/notebook used to obtain accuracy reported above along with details of how to use it and reproduce accuracy.
<hr>

## References
Link to references
<hr>

## Contributors
Contributors' name
<hr>

## License
Add license information
<hr>

### CODE REVIEW

File: emotion_ferplus/README.md [hunk 1]
```diff
@@ -33,12 +33,30 @@ import numpy as np
...
+      if img_arr.shape[2] == 3 or img_arr.shape[2] == 4 :
+          for i in range(64):
+            for j in range(64):
+                rgb = img_arr[i][j]
+                new_arr[i][j] = (rgb[0] * 0.299 + rgb[1] * 0.587 + rgb[2] * 0.114  - 127.5)/127.5
+  # if one channel
+  else:
+      for i in range(64):
+        for j in range(64):
+            new_arr[i][j] = (img_arr[i][j]  - 127.5)/127.5
```
Comment (ebarsoum) [inline emotion_ferplus/README.md]: The model already does the normalization, so this will actually break.

File: emotion_ferplus/README.md [hunk 2]
```diff
@@ -33,12 +33,30 @@ import numpy as np
 from PIL import Image
 
 def preprocess(image_path):
+  # model input shape
   input_shape = (1, 1, 64, 64)
+  # load image and resize
   img = Image.open(image_path)
```
Comment (ebarsoum) [inline emotion_ferplus/README.md]: You can convert  to gray level by calling convert() on Image object no need to iterate.: img = Image.open('image.png').convert('L')

**Comment by ebarsoum** [comment]
No update for this PR.
